src/vsr/message_header.zig

```
@@ -553,9 +553,7 @@ pub const Header = extern struct {
                     if (self.parent != 0) return "register: parent != 0";
                     if (self.session != 0) return "register: session != 0";
                     if (self.request != 0) return "register: request != 0";
-                    if (self.size != @sizeOf(Header) and // Compat(v0.15.3)
-                        self.size != @sizeOf(Header) + @sizeOf(vsr.RegisterRequest))
-                    {
+                    if (self.size != @sizeOf(Header) + @sizeOf(vsr.RegisterRequest)) {
```

Huh, this is a puzzle! If we do check this here, then we won't send an eviction message to the client, and instead silently drop the message. We do `.invalid` check before we do client_version_min check.

What are we to do? 🤔 

I suggest the following:

* Keep this code as is for at least a couple of releases more, such that, if people don't get our deprivation memo, they get a clean error message, rather than a silent no-answer
* But remove it afterwards: I think there's a lot of value in not admitting invalid values into the system, and, once almost everyone migrates, we shouldn't loose much from omitting the eviction message.

My hypothesis is that we won't be doing many similar "frame changes" in the future. If that turns out to be wrong, we need to change on message to move client version check before header validity check, but I think we won't need it, or at least won't need it yet!